### ISSUE TITLE
Unable to extend more than one Immutable builder in @Enclosing (in Eclipse)

### ISSUE THREAD

**Issue by pabad** [immutables/immutables#270]
We have been getting an error about a circular dependency in Eclipse for a system that builds correctly using gradle.

So far we have narrowed the case to having two immutables defined inside an enclosing and having both define a builder extending the immutable implementation:
```java
@Value.Enclosing
final class Test {
	@Value.Immutable
	static abstract class Type1 {
		abstract int a();
		abstract int b();

		static class Builder extends ImmutableTest.Type1.Builder { }
	}

	@Value.Immutable
	static abstract class Type2 {
		abstract int c();
		abstract int d();

		static class Builder extends ImmutableTest.Type2.Builder { }
	}
}
```

One of the builders is created correctly, while the other not. The immutable implementation ends up like this:
```java
final class ImmutableTest {
  static final class Type1 extends Test.Type1 { ... 
    ..
    static final class Builder extends Test.Type1.Builder { ...}
  }
  static final class Type2 extends Test.Type2 {
    ...
    static class Builder { ... }
}
```

Now, as I said, this is building correctly in gradle, but fails always in eclipse. Could it be to the way eclipse uses annotation processor? Or perhaps you are downcasting to javac specific classes (I've seen this in far too many processors) and it behaves differently when using ejc?

At eclipse's log I could only find this relevant entry:
```
Generated source file name collission. Attempt to overwrite already generated file: ImmutableTest, javax.annotation.processing.FilerException: Source file already created: /repro/.apt_generated/ImmutableTest.java
```

**Comment by elucash** [comment]
We are downcasting to both Javac and Eclipse classes in order to overcome many bugs, and this might be one of them. I was able to reproduce it so I hope it fixable

**Comment by elucash** [comment]
Released in 2.1.12. Thank you!
